Props based attributes

OlehRb opened the pull request:
Resolves https://github.com/krakenjs/zoid/issues/308 via adding a possibility to define `attributes` field in the same way as we do for `url` field

bluepnume commented:
Code looks great, thanks for adding this! Would you mind adding a quick test to make sure attributes are set correctly in both cases? Thanks!

OlehRb commented:
@bluepnume , it seems like my original approach does not work for dynamic attributes, because it looks like the `props` variable gets mutated somewhere along the way..

So I've updated the computation of dynamic attributes to be performed in a dedicated function (the same way as it is implemented for `url` parameter)

Also I've added a few tests for dynamic and static attributes
Please, let me know if you have any concerns about current implementation or if I can improve / extend test cases in any way (cause to be honest, I don't have too much of experience with tests).
